test/e2e/framework/pod/wait.go

```diff
@@ -686,6 +690,66 @@ func WaitForContainerRunning(c clientset.Interface, namespace, podName, containe
 	})
 }
 
+// WaitForPodStatusByInformer waits pod status change by informer
+func WaitForPodStatusByInformer(c clientset.Interface, podNamespace, podName string, timeout time.Duration, condition podCondition) error {
```

Comment:
@pohly @tallclair I wonder if we should move all these functions in this file to use an listWatch instead of a poller instead of start duplicating them, this way we can reuse them.

Right now, some tests are already using a list-watch embedding the logic within the test

an example of an e2e test that embeds the list-watch logic is here 
https://github.com/kubernetes/kubernetes/blob/6035ccd46c28b4f000259eb76337557cec6a50f1/test/e2e/common/pods.go#L240-L335